CRM/Custom/Page/CustomDataTrait.php

```diff
@@ -35,15 +35,21 @@ trait CRM_Custom_Page_CustomDataTrait {
    *
    */
   protected function getCustomDataFieldsForEntityDisplay(string $entity, int $id): array {
-    $values = civicrm_api4($entity, 'get', [
-      'select' => [
-        'custom.*',
-      ],
-      'where' => [
-        ['id', '=', $id],
-      ],
-      'checkPermissions' => TRUE,
-    ])->single();
+    try {
+      $values = civicrm_api4($entity, 'get', [
+        'select' => [
+          'custom.*',
+        ],
+        'where' => [
+          ['id', '=', $id],
+        ],
+        'checkPermissions' => TRUE,
+      ])->single();
+    }
+    catch (CRM_Core_Exception $e) {
+      Civi::log()->warning('Tried loading Custom field data for Entity ' . $entity . ' id: ' . $id . ' and got the following message . ' $e->getMessage());
```

Comment:
Quote is in the wrong place. should be `' and go the following message: ' . $e->getMessage()`

Reply:
This makes me think that this has not been tested at all. This gives a parse error, even when the exception is not caught.

Reply:
We're currently running this patch in production (with the parse error corrected).

Prior to the implementation of this patch, we had many reports of "Expected one Address but found 0" occurring intermittently (as in, people who experienced this error upon loading a contact were able to load it without issue later in the day).

There has not been such a report since, and I can see quite a number of instances where the new logs appearing.

It's important to understand that we do not have any custom fields associated with Addresses or Emails, so the fact that this function simply returns with an empty array doesn't seem to have any impact on the operation of our site.